canonicalwebteam/blog_extension/logic.py

```diff
@@ -0,0 +1,156 @@
+    image scaling
+
+    :param content: The HTML string to convert
+
+    :returns: Update HTML string with converted images
+    """
+    cloudinary = "https://res.cloudinary.com/"
+
+    urls = [
+        cloudinary + "canonical/image/fetch/q_auto,f_auto,w_350/\g<url>",
+        cloudinary + "canonical/image/fetch/q_auto,f_auto,w_650/\g<url>",
+        cloudinary + "canonical/image/fetch/q_auto,f_auto,w_1300/\g<url>",
+        cloudinary + "canonical/image/fetch/q_auto,f_auto,w_1950/\g<url>",
+    ]
+
+    image_match = (
+        r'<img(?P<prefix>[^>]*) src="(?P<url>[^"]+)"(?P<suffix>[^>]*)>'
+    )
+    replacement = (
+        "<img\g<prefix>"
+        f' decoding="async"'
+        f' src="{urls[1]}"'
+        f' srcset="{urls[0]} 350w, {urls[1]} 650w, {urls[2]} 1300w,'
+        f' {urls[3]} 1950w"'
+        f' sizes="(max-width: 400px) 350w, 650px"'
+        "\g<suffix>>"
+    )
+
+    return re.sub(image_match, replacement, content)
+
+
+def transform_article(
+    article, featured_image=None, author=None, optimise_images=False
+):
+    """Transform article to include featured image, human readable
+    date and a stipped version of the excerpt
+
+    :param article: The raw article object
+    :param featured_image: The featured image string
+
+    :returns: The transformed article
+    """
+    article["image"] = featured_image
+
+    article["author"] = author
+
+    if "date_gmt" in article:
+        article_gmt = article["date_gmt"]
+        article_date = datetime.strptime(article_gmt, "%Y-%m-%dT%H:%M:%S")
+        article["date"] = article_date.strftime("%-d %B %Y")
+
+    if "excerpt" in article and "rendered" in article["excerpt"]:
+        article["excerpt"]["raw"] = strip_excerpt(
+            article["excerpt"]["rendered"]
+        )[:340]
+
+        # If the excerpt doesn't end before 340 characters, add ellipsis
+        raw_article = article["excerpt"]["raw"]
+        # split at the last 3 characters
+        raw_article_start = raw_article[:-3]
+        raw_article_end = raw_article[-3:]
+        # for the last 3 characters replace any part of […]
+        raw_article_end = raw_article_end.replace("[", "")
+        raw_article_end = raw_article_end.replace("…", "")
+        raw_article_end = raw_article_end.replace("]", "")
+        # join it back up
+        article["excerpt"]["raw"] = "".join(
+            [raw_article_start, raw_article_end, " […]"]
+        )
+
+    if (
+        optimise_images
+        and "content" in article
+        and "rendered" in article["content"]
+    ):
+        article["content"]["rendered"] = replace_images_with_cloudinary(
+            article["content"]["rendered"]
+        )
+
+    return article
+
+
+def change_url(feed, host):
+    """Change insights urls to snapcraft.io/blog
+
+    :param feed: String with urls
+
+    :returns: A string with converted urls
+    """
+    url_regex = re.compile(
+        "https:\/\/admin.insights.ubuntu.com(\/\d{4}\/\d{2}\/\d{2})?"
```

Comment:
As mentioned in snapcraft.io, I suspect you don't need to escape the slashes here.

Reply:
yeah I was wainting to fix on s.io before applying the same configuration

Reply:
Moved to an issue: https://github.com/canonical-webteam/blog-flask-extension/issues/7